### PR TITLE
Adding Azure RetryPolicy support

### PR DESCRIPTION
**Pull request by awlx**
This adds support for the most important RetryPolicy variables.

See https://azuresdkdocs.blob.core.windows.net/$web/python/azure-core/1.9.0/azure.core.pipeline.policies.html?highlight=retrypolicy#azure.core.pipeline.policies.RetryPolicy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,13 @@ providers:
     # Resource Group name:
     resource_group: 'TestResource1'
     # All are required to authenticate.
+    # Azure RetryPolicy Settings
+    # https://azuresdkdocs.blob.core.windows.net/$web/python/azure-core/1.9.0/azure.core.pipeline.policies.html?highlight=retrypolicy#azure.core.pipeline.policies.RetryPolicy
+    # Total_retries default 10
+    client_total_retries: 10
+    # status_retries default 3
+    client_status_retries: 3
```
Comment (ross) [inline README.md:59]: Super minor, but the convention for whatever reason has been to comment out optional stuff in these.

```suggestion
    #client_total_retries: 10
    # status_retries default 3
    #client_status_retries: 3
```

**Review by ross — COMMENTED** [review]
LGTM. One minor change set and it should be good to 🚢

**Review by ross — APPROVED** [review]

**Comment by ross** [comment]
Looks like some `./script/lint` objections.

```
octodns_azure/__init__.py:507:80: E501 line too long (100 > 79 characters)
octodns_azure/__init__.py:527:45: E126 continuation line over-indented for hanging indent
octodns_azure/__init__.py:528:80: E501 line too long (81 > 79 characters)
octodns_azure/__init__.py:529:41: E126 continuation line over-indented for hanging indent
```

File: octodns_azure/__init__.py [hunk 2]
```diff
@@ -539,6 +545,7 @@ def _dns_client(self):
                     logger=logger,
                 ),
                 subscription_id=self._dns_client_subscription_id,
+                retry_policy=self._dns_client_retry_policy
```
Comment (viranch) [inline octodns_azure/__init__.py:548]: Should we also set equivalent of this for `self.__tm_client`?

File: README.md [hunk 3]
```diff
@@ -51,6 +51,13 @@ providers:
     # Resource Group name:
     resource_group: 'TestResource1'
     # All are required to authenticate.
+    # Azure RetryPolicy Settings
+    # https://azuresdkdocs.blob.core.windows.net/$web/python/azure-core/1.9.0/azure.core.pipeline.policies.html?highlight=retrypolicy#azure.core.pipeline.policies.RetryPolicy
+    # Total_retries default 10
+    #client_total_retries: 10
+    # status_retries default 3
+    #client_status_retries: 3
+    # All are optional
```
Comment (viranch) [inline README.md:60]: Can this be put above at line 54?

**Comment by awlx** [comment]
Hoops, but how did I pass the local linter 🤔?

Will check and fix it of course :).

File: octodns_azure/__init__.py (hunk 2 above)
Comment (awlx) [reply octodns_azure/__init__.py:548]: Let me look into it :). Also thank you both for reviewing already!

Comment (awlx) [reply octodns_azure/__init__.py:548]: It looks like it doesn't implement the retry_policy? 🤔 
https://docs.microsoft.com/en-us/python/api/azure-mgmt-trafficmanager/azure.mgmt.trafficmanager.trafficmanagermanagementclient?view=azure-python

Comment (viranch) [reply octodns_azure/__init__.py:551]: Looking at the code, the `TrafficManagerClient` does take `kwargs` (annoying that its not documented) which gets funneled to a chain of various other objects internally that are indeed respecting `retry_policy`. Even if not, by virtue of `kwargs`, an unknown key will just get ignored, so passing the same `retry_policy` to `__tm_client` sounds harmless.

I'd prefer to keep stuff uniform across the various `*Client` objects as much as possible.

Comment (awlx) [reply octodns_azure/__init__.py:551]: Sounds good to me. Will add it 👍

Comment (awlx) [reply octodns_azure/__init__.py:551]: Added via 7b413ac

**Review by ross — APPROVED** [review]
👍, will leave the final ✅ & merge to @viranch since he's much more familiar with the provider. 

Thanks @awlx

**Review by viranch — APPROVED** [review]
Good to go, thanks @awlx!

**Pull request merged**